X2WOTCCommunityHighlander/Src/XComGame/Classes/XComGameState_Unit.uc

```diff
@@ -6604,73 +6611,71 @@ protected function OnUnitDied(XComGameState NewGameState, Object CauseOfDeath, c
 
 				CheckForFlankingEnemyKill(NewGameState, Killer);
 
-				// Issue #562 - Allow mods to cap kill XP
-				/// HL-Docs: ref:CanAwardKillXp
-				if (TriggerCanAwardKillXp(Killer))
+				// If the Wet Work GTS bonus is active, increment the Wet Work kill counter
+				XComHQ = XComGameState_HeadquartersXCom(History.GetSingleGameStateObjectForClass(class'XComGameState_HeadquartersXCom', true));
+				if(XComHQ != none)
 				{
-					// If the Wet Work GTS bonus is active, increment the Wet Work kill counter
-					XComHQ = XComGameState_HeadquartersXCom(History.GetSingleGameStateObjectForClass(class'XComGameState_HeadquartersXCom', true));
-					if(XComHQ != none)
+					if(XComHQ.SoldierUnlockTemplates.Find('WetWorkUnlock') != INDEX_NONE)
 					{
-						if(XComHQ.SoldierUnlockTemplates.Find('WetWorkUnlock') != INDEX_NONE)
-						{
-							WetWorkXp = 1;  // Issue #562: store in a variable so it can be overridden
-						}
-
-						BonusKillXp = XComHQ.BonusKillXP;  // Issue #562: store in a variable so it can be overridden
+						WetWorkXp = 1;  // Issue #562: store in a variable so it can be overridden
 					}
 
-					// Start Issue #562
-					/// HL-Docs: ref:OverrideKillXp
-					// Get the default values for kill XP and kill assist XP and then allow mods to
-					// override these values before applying them to the unit.
-					KillXp = GetMyTemplate().KillContribution; // Allows specific units to contribute different amounts to the kill total
-					KillAssistXp = KillXp;
+					BonusKillXp = XComHQ.BonusKillXP;  // Issue #562: store in a variable so it can be overridden
+				}
+
+				// Start Issue #562
+				/// HL-Docs: ref:OverrideKillXp
+
+				// Undo xp from #221
+				Killer.KillCount -= KillXp;
```

Comment:
I honestly don't think backwards compatibility is worth the confusion here, particularly as the kill XP is awarded even if `CanEarnXp()` returns `false` (which shouldn't happen in my view). I don't think #221 should ever have incremented `KillCount` outside of the `CanEarnXp()` check and if we ever want to evolve the hooks for XP gain, this special handling will just be a source of new bugs.